### DISCUSSION TITLE
Обход youtube

### DISCUSSION THREAD

**Discussion by Waujito** [Waujito/youtubeUnblock#173] [General]
Сейчас ютуб должен работать на дефолтных настройках. 

Если возникают проблемы с работоспособностью, советую убедиться, что youtubeUnblock ведёт лог о том, что видит ютуб.
Если в логах пусто - запустите его с флагом `--trace` и посмотрите, захватывает ли он трафик в принципе.

Советую отключить kyber и quic в браузере, чтобы точно ничего не мешало работе. Если возникают проблемы на TV или мобильных устройствах, можно добавить флаг `--quic-drop`, чтобы блокировать весь quic на уровне youtubeUnblock.

Если вы уверены, что проблема именно на стороне обхода, советую попробовать эти флаги:
```
youtubeUnblock --fake-sni-type=custom --fake-custom-payload=00 --fake-sni-seq-len=10 --frag-sni-faked=1 --frag-sni-reverse=0 --frag-sni-pos=1 --frag-middle-sni=0 --seg2delay=100
```

Можете писать сюда вопросы по работе ютуба на ваших устройствах и провайдерах.

**Comment by irulanCorrino** [comment]
hi, Waujito! вчера перестал работать ютюб. и паттерн появления проблем был похож на внешнее воздействие. до 0:00 6 Дек. всё работало. с 0:00 сегодня клипы грузятся до 40-60сек. потом колесо вечности...
и вначале я смотрела клипы Sevdaliza 'Martyr' и Melanie Martinez ' Spider Web' (всё было OK) потом перестали грузиться клипы (Charli XCX, etc.) но я смогла вернуться в браузере (firefox) и снова полностью посмотреть тот клип Севдализы.
у `clipgrab` котрый я использую постоянно нет проблем со скачиванием клипов без всякого включения youtubeUnblock (в отличии от того что было в сентябре).
youtubeUnblock не помогает (использую модуль для ядра archlinux (artix), обновила 5мин. назад).
и ещё деталь --ютюб врёт про моё время просмотра --только два клипа были просмотренны полностью (сегодня до возникновения проблемы): 'Martyr' и 'Spider Web' (они переместились вверх в истории после того как я пыталась смотреть их сейчас в 17:00 --теперь тоже не полностью, конечно же)
![faked watchtime Screenshot_2024-12-06_16-33-01](https://github.com/user-attachments/assets/e3f7d885-212f-4f64-8897-ecc160f6b635)

**Reply by Waujito** [reply]
Может быть, проблема в ipv6? Если он включен, попробуйте его отключить. Также попробуйте поставить классический userspace анблок и запустить его с флагами `--sni-domains=all --exclude-domains=.ru --quic-drop --fbegin`. Отмечу, что лично у меня недавно после обновления в фаерфоксе включился kyber. Отключите его.

**Reply by irulanCorrino** [reply]
нет-нет. я забыла дополнить мой рапорт следующим критическим фактом --перед тем как запустить мою artix linux систему и начать смотреть тот клип Севдализы после полуночи, я безуспешно пыталась смотреть ютюб с моего Android смартфона (через домашний wifi, тот же что и для моей artix).
на скриншоте который я вставила в предыдущее сообщение этот клип виден под 'Yesterday': Nicki Minaj 'Last Time I Saw You'.
sorry for misleading you, Waujito.


у меня есть гипотеза (конспирологическая): `clipgrab` работает но смарфон нет --я не могу это объяснить ...man-in-the-middle аттака на ютюб с подменой статуса просмотра видео (`completed`) чтобы Гугл кэш сервер перестал обслуживать запросы ютюб клиента ...но тогда СПТУ должно декодировать https траффик, как я понимаю, так? но тогда это ОЧЕНЬ дорогое СПТУ.
но после 5 декября ВСЕ видео останавливаються на 59сек.

я отключила кибер в браузере (у меня тоже было это обновление) --не помогло.
я попробую youtubeUnblock в том режиме который вы рекоммендуете.
о проблеме в ipv6 я не знаю но раньше у меня всегда было ipv4 и ipv6 было неактивно (но я могу запутаться с пониманием всех этих сетевых настроек, я интерфейс дизайнер и сейчас переучиваюсь на C language & DSP programming).

у меня есть ещё одна проблема в браузере: `DNS over HTTPS` я не могу это настроить [BAD_URL].
не помню как но у меня было установлено `custom` (с тем самым сообщением ошибки), я проверила --это был яндекс (который я просто ненавижу):
`Location: Moscow, Russian Federation - 77.88.8.88 is a IP address allocated to Yandex LLC.`
я переключила на `default` Cloudflare DNS и оно активно но не помогает никак.

**Reply by irulanCorrino** [reply]
если вы мне объясните как использовать `wireshark` для дебагга моей сети, я с удовольствием последую вашим подробным инструкциям.
даже могу прошивку на роутере сменить :)

**Reply by irulanCorrino** [reply]
no no. it is a bug at youtube probably. i have found it exists for a few months at least and is not related to russian censure.
people around internet speculate google had broken their API themselves in a pursuit of their ads business.

**Comment by LutikovPavel** [comment]
Здравствуйте, роутер Xiaomi AX 3000T OpenWrt 24.10.1. На дефолтных настройках Youtube работает только по WiFi, когда подключаю ноут по проводу, страницу грузит но видео в бесконечной загрузке, подскажите в чём может быть проблема?
